### ISSUE TITLE
Poor eyebrowse/perspective integration

### ISSUE THREAD

**Issue by darkfeline** [syl20bnr/spacemacs#5871]
Example:

1. Open file foo/a in persp 1 workspace 1
2. Open file foo/b in persp 1 workspace 2
3. Open file bar/b in persp 2 workspace 2 (`SPC p l`)
4. Switch to persp 2 workspace 1: **layout for project bar has foo/a**
5. Open bar/a in persp 2 workspace 1
6. Switch to persp 1 workspace 1: **workspace 1 is now foo/b, not foo/a**
7. Switch to persp 1 workspace 2: **layout for project foo has bar/a**

Basically eyebrowse and perspectives have zero integration, they simply coexist in a passive-aggressive manner.  As it is, it's very difficult to use both eyebrowse and perspectives in any meaningful sense.

**Comment by syl20bnr** [comment]
I can confirm this behaviour. To be more accurate than `poor integration` :-) the issue is that there is no workspace isolation per layout, workspaces are shared except for the current one. I believe that this is a known limitation of the current implementation. cc @bmag

**Comment by bmag** [comment]
IIRC there should be workspace isolation. I'll look into.

**Comment by bmag** [comment]
Got it. A hook was renamed upstream, see #5874

**Comment by CandyGumdrop** [comment]
Wil l this fix be brought into master any time soon?  It works great on develop now, but master is kind of unusable for me right now because of this.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
